src/backend/distributed/transaction/worker_transaction.c

```diff
@@ -329,6 +328,65 @@ SendCommandToMetadataWorkersParams(const char *command,
 }
 
 
+/*
+ * SendCommandToWorkersOptionalInParallel sends the given command to workers in parallel.
+ * It does error if there is a problem while sending the query, but it doesn't error
+ * if there is a problem while executing the query.
+ */
+void
+SendCommandToWorkersOptionalInParallel(TargetWorkerSet targetWorkerSet, const
+									   char *command,
+									   const char *user)
+{
+	List *connectionList = NIL;
+	ListCell *connectionCell = NULL;
+	List *workerNodeList = TargetWorkerSetNodeList(targetWorkerSet, ShareLock);
+	ListCell *workerNodeCell = NULL;
+
+
+	/* open connections in parallel */
+	foreach(workerNodeCell, workerNodeList)
+	{
+		WorkerNode *workerNode = (WorkerNode *) lfirst(workerNodeCell);
+		char *nodeName = workerNode->workerName;
+		int nodePort = workerNode->workerPort;
+		int32 connectionFlags = 0;
+
+		MultiConnection *connection = StartNodeUserDatabaseConnection(connectionFlags,
+																	  nodeName, nodePort,
+																	  user, NULL);
+		connectionList = lappend(connectionList, connection);
+	}
+
+	/* finish opening connections */
+	FinishConnectionListEstablishment(connectionList);
+
+	RemoteTransactionsBeginIfNecessary(connectionList);
+
+	/* send commands in parallel */
+	foreach(connectionCell, connectionList)
+	{
+		MultiConnection *connection = (MultiConnection *) lfirst(connectionCell);
+
+		int querySent = SendRemoteCommand(connection, command);
+		if (querySent == 0)
+		{
+			ReportConnectionError(connection, ERROR);
+		}
+	}
+
+	/* get results */
+	foreach(connectionCell, connectionList)
+	{
+		MultiConnection *connection = (MultiConnection *) lfirst(connectionCell);
+
+		PGresult *result = GetRemoteCommandResult(connection, true);
+		PQclear(result);
+		CloseConnection(connection);
```

Comment:
This connection might be used for other things (incl. uncommitted writes), so it should not be closed.

However, we probably should have a ClearResults.

Reply:
probably ForgetResults is better here. However it seems that while we are cleaning the results we read from connections sequentially. Ideally it would be better to not read sequentially but in parallel especially when we dont even care about the results?